cmd/generate-workflow.go

```diff
@@ -74,44 +57,67 @@ func init() {
 	rootCmd.AddCommand(newGenerateWorkflowCmd())
 }
 
-func (gwc *generateWorkflowCmd) generateWorkflows(dest string, deployType string, flagVariables []string, templateWriter templatewriter.TemplateWriter, flagValuesMap map[string]string) error {
-	if flagValuesMap == nil {
-		return fmt.Errorf("flagValuesMap is nil")
-	}
+func (gwc *generateWorkflowCmd) generateWorkflows() error {
 	var err error
-	for _, flagVar := range flagVariables {
-		flagVarName, flagVarValue, ok := strings.Cut(flagVar, "=")
-		if !ok {
-			return fmt.Errorf("invalid variable format: %s", flagVar)
-		}
-		flagValuesMap[flagVarName] = flagVarValue
-		log.Debugf("flag variable %s=%s", flagVarName, flagVarValue)
-	}
 
-	if deployType == "" {
-		selection := &promptui.Select{
-			Label: "Select k8s Deployment Type",
-			Items: []string{"helm", "kustomize", "manifests"},
-		}
+	flagVariablesMap = flagVariablesToMap(gwc.flagVariables)
 
-		_, deployType, err = selection.Run()
-		if err != nil {
-			return err
+	if gwc.deployType == "" {
+		if flagValue := flagVariablesMap["deploy-type"]; flagValue == "helm" || flagValue == "kustomize" || flagValue == "manifests" {
```

Comment:
these would be better stored as exported enums since we're using them repeatedly

Reply:
additionally, we can likely detect the directories we use and if only one is present, default to that deploytype eg ./charts ./manifests and ./overlays

i'll make an issue to track it so we can do that in another PR

Reply:
moved to https://github.com/Azure/draft/issues/332

Reply:
I'm actually going to take this logic out since the new implementation of the variables []string flag is that it's for template variables